### PR TITLE
Error handle for NaN values

### LINKED ISSUE #182: LogMetricsAsync does not handle NaN values
**Describe the bug**
Its possible some metrics return a NaN. For example Clustering returns a value for NormalizedMutualInformation only if the label column name is used.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.clusteringmetrics.normalizedmutualinformation?view=ml-dotnet#Microsoft_ML_Data_ClusteringMetrics_NormalizedMutualInformation

**To Reproduce**
1. Go to MLOps.NET.Clustering
2. Change             
  var metrics = mlContext.Clustering.Evaluate(predictions, "Label", "Score", "Features");
to more commonly done;
   var metrics = mlContext.Clustering.Evaluate(predictions);
3. Run

**Expected behavior**
LogMetricsAsync handles NaN and ignore it.

**Screenshots**
image

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**
Found during creation of #181

### PR DESCRIPTION
**Pull request by Brett-Parker**
added a catch for NaN value returned for metrics

## Resolves
Fixes #182 

## Description
Error handles NaN values that are set to LogMetricsAsync - E.G Clustering returns a value for NormalizedMutualInformation only if the label column name is used.

### CODE REVIEW

**Comment by Brett-Parker** [comment]
Please advise if you would have handled this differently.

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs [hunk 1]
```diff
@@ -49,7 +49,10 @@ public async Task LogMetricAsync(Guid runId, string metricName, double metricVal
 
             foreach (var metric in properties)
             {
-                await LogMetricAsync(runId, metric.Name, (double)metric.GetValue(metrics));
+                if (!(double.IsNaN((double)metric.GetValue(metrics))))
```
Comment (aslotte) [inline src/MLOps.NET/Catalogs/EvaluationCatalog.cs:52]:
```suggestion
               var value = (double)metric.GetValue(metrics);
                if (double.IsNaN(metric)) continue;
```

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs [hunk 2]
```diff
@@ -49,7 +49,10 @@ public async Task LogMetricAsync(Guid runId, string metricName, double metricVal
...
+                {
+                    await LogMetricAsync(runId, metric.Name, (double)metric.GetValue(metrics));
```
Comment (aslotte) [inline src/MLOps.NET/Catalogs/EvaluationCatalog.cs:54]:
```suggestion
                    await LogMetricAsync(runId, metric.Name, metric);
```

**Comment by aslotte** [comment]
This is great @Brett-Parker! Added two minor suggestions for readability, let me know if you agree.

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs (hunk 1 above)
Comment (Brett-Parker) [reply src/MLOps.NET/Catalogs/EvaluationCatalog.cs:52]: Should - if (double.IsNaN(metric)) continue;

not be this - if (double.IsNaN(value)) continue;

@aslotte

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs (hunk 2 above)
Comment (Brett-Parker) [reply src/MLOps.NET/Catalogs/EvaluationCatalog.cs:54]: Should this - await LogMetricAsync(runId, metric.Name, metric);

not be this - await LogMetricAsync(runId, metric.Name, value);

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs (hunk 1 above)
Comment (aslotte) [reply src/MLOps.NET/Catalogs/EvaluationCatalog.cs:52]: Great catch, yeap!

File: src/MLOps.NET/Catalogs/EvaluationCatalog.cs (hunk 2 above)
Comment (aslotte) [reply src/MLOps.NET/Catalogs/EvaluationCatalog.cs:54]: Totally correct, updated my suggestion.

**Comment by Brett-Parker** [comment]
@aslotte I have now closed the review comments as resolved. I am now unable to review or merge. Thanks :)

**Comment by Brett-Parker** [comment]
@aslotte ready for review. Stupidly committed with files from another issue :(

Removed them and made changes as discussed.

**Comment by Brett-Parker** [comment]
@aslotte should we handle NaN values for LogMetricAsync as the user could pass them individually. They will get the standard exception.

**Comment by aslotte** [comment]
It doesn't hurt to check it but I honestly don't think we need it there. If a user passes it intentionally then I think it will (hopefully) be pretty straightforward to understand what the exception means. I leave it up to you to decide :)

**Pull request merged**